src/main/java/nextstep/laddergame/LineGenerator.java

```diff
@@ -0,0 +1,36 @@
+package nextstep.laddergame;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+import nextstep.laddergame.domain.Line;
+
+public class LineGenerator {
+
+  private final int countOfMembers;
+
+  public LineGenerator(int countOfMembers) {
+    this.countOfMembers = countOfMembers;
+  }
+
+  public Line generate(DrawStrategy strategy) {
+    List<Boolean> points = new ArrayList<>();
+
+    IntStream.range(0, countOfMembers)
+        .mapToObj(index -> drawLine(isRowContinuousTrue(points, index), strategy))
+        .forEach(points::add);
+
+    return new Line(points);
```

Comment:
```suggestion
    return Stream.generate(strategy::isLine)
                .limit(countOfMembers)
                .collect(Collector.of(ProtoLine::new,
                                      ProtoLine::add,
                                      ProtoLine::merge,
                                      ProtoLine::toLine));
```

이렇게 바꿔볼 수 있습니다. 이때 사용할 `ProtoLine` 은 다음과 같습니다.
```java
class ProtoLine {
    private final List<Boolean> points;

    private ProtoLine(List<Boolean> points) {
        this.points = points;
    }

    public ProtoLine() {
        this(new ArrayList<>());
    }

    protected void add(Boolean next) {
        if (drewLeft()) {
            points.add(false);
            return;
        }

        points.add(next);
    }

    protected ProtoLine merge(ProtoLine other) {
        List<Boolean> result = new ArrayList<>(points);
        result.addAll(other.points);
        return new ProtoLine(result);
    }

    protected Line toLine() {
        return new Line(points);
    }

    private Boolean drewLeft() {
        return !points.isEmpty() && points.get(points.size() - 1);
    }
}
```

Reply:
조언 감사합니다!! 해당 방식대로 수정하겠습니다.